### PR TITLE
Synchronize the player's food list from the solcarrot mod.

### PR DESCRIPTION
**Pull request by bluelightning32**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -5,10 +5,16 @@ repositories {
...
+        url = "https://minecraft.curseforge.com/api/maven/"
+    }
 }
 
 dependencies {
     compileOnly "org.spongepowered:spongecommon:${sponge_version}:dev"
 
     compileOnly "com.azanor.baubles:Baubles:${baubles_version}"
+
+    deobfCompile "spice-of-life-carrot-edition:solcarrot:${solcarrot_version}"
```
Comment (BrainStone) [inline build.gradle]: Use `deobfProvided` as that indicates an optional dependency.

File: src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java [hunk 2]
```diff
@@ -58,6 +60,9 @@
               enderChestSyncModule,
               gameModeSyncModule,
               experienceSyncModule,
+              // Sync the solcarrot data before syncing the player's current health, because the
+              // solcarrot sets the player's maxhealth.
+              solCarrotSyncModule,
```
Comment (BrainStone) [inline src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java]: Did you test that? Like does it break stuff if it happens after syncing the health?

File: src/main/java/world/jnc/invsync/util/serializer/module/mod/SolCarrotSyncModule.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  private static class Helper {
+    private static final DataQuery STACK = DataQuery.of("stack");
+    private static final DataQuery META = DataQuery.of("meta");
+
+    private static DataView serialize(Player player, DataView container) {
+      FoodCapability foodCapability =
+          NativeInventorySerializer.getNativePlayer(player)
+              .getCapability(FoodCapability.FOOD_CAPABILITY, null);
+      List<DataView> serializedFoods = new LinkedList<>();
+      if (foodCapability != null) {
```
Comment (BrainStone) [inline src/main/java/world/jnc/invsync/util/serializer/module/mod/SolCarrotSyncModule.java]: Blank lines haven't killed anyone yet ;)

Have a little look at the rest of the code to see what spacing style I usually use.

And lastly if you haven't done that yet (though it looks like you did) run `./gradlew spotlessApply` to have correct formatting.

File: src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java (hunk 2 above)
Comment (bluelightning32) [reply src/main/java/world/jnc/invsync/util/serializer/PlayerSerializer.java]: Yes, I tested it. Here's what can happen if they run out of order:
1. The player has a current health of 40 and max health of 40 in the database.
2. The player.dat is blank, so the player starts off with 10 current health and 10 max health.
3. The 40 current health is restored, but it's above the max of 10, so it gets capped at 10.
4. The max health of 40 is restored, but the current health was already set to 10.

File: src/main/java/world/jnc/invsync/util/serializer/module/mod/SolCarrotSyncModule.java (hunk 3 above)
Comment (bluelightning32) [reply src/main/java/world/jnc/invsync/util/serializer/module/mod/SolCarrotSyncModule.java]: I added some new lines. Let me know if it still doesn't match your style.

Yes, I ran `gradlew spotlessApply`. That's a nice tool. I didn't know about it until it printed warnings.

**Pull request merged**